Wikipedia/Code/WMFArticlePreviewTableViewCell.m

```diff
@@ -30,11 +34,22 @@ @implementation WMFArticlePreviewTableViewCell
 
 #pragma mark - Setup
 
+- (void)blurAndShowLoadingIndicator {
+    self.blurView.hidden = NO;
+    [self.activityIndicator startAnimating];
+}
+
+- (void)unblurAndHideLoadingIndicator {
+    self.blurView.hidden = YES;
+    [self.activityIndicator stopAnimating];
```

Comment:
tip: can also show/hide a container view instead of toggling the visibility of a group of subviews. in this case: the spinner is always spinning when shown, and it always needs to be shown on top of the blur. can just group blur & spinner in a container which we show/hide (or even constrain) as needed.